packages/medusa/src/api/routes/admin/orders/update-claim.ts

```diff
@@ -83,48 +96,107 @@ import { defaultRelations, defaultFields } from "./"
 export default async (req, res) => {
   const { id, claim_id } = req.params
 
-  const schema = Validator.object().keys({
-    claim_items: Validator.array()
-      .items({
-        id: Validator.string().required(),
-        note: Validator.string().allow(null, ""),
-        reason: Validator.string().allow(null, ""),
-        images: Validator.array().items({
-          id: Validator.string().optional(),
-          url: Validator.string().optional(),
-        }),
-        tags: Validator.array().items({
-          id: Validator.string().optional(),
-          value: Validator.string().optional(),
-        }),
-        metadata: Validator.object().optional(),
-      })
-      .optional(),
-    shipping_methods: Validator.array()
-      .items({
-        id: Validator.string().optional(),
-        option_id: Validator.string().optional(),
-        price: Validator.number().integer().optional(),
-      })
-      .optional(),
-    no_notification: Validator.boolean().optional(),
-    metadata: Validator.object().optional(),
-  })
-
-  const { value, error } = schema.validate(req.body)
-  if (error) {
-    throw new MedusaError(MedusaError.Types.INVALID_DATA, error.details)
-  }
+  const validated = await validator(
+    AdminPostOrdersOrderClaimsClaimReq,
+    req.body
+  )
 
-  const orderService = req.scope.resolve("orderService")
-  const claimService = req.scope.resolve("claimService")
+  const orderService: OrderService = req.scope.resolve("orderService")
+  const claimService: ClaimService = req.scope.resolve("claimService")
 
-  await claimService.update(claim_id, value)
+  await claimService.update(claim_id, validated)
 
   const data = await orderService.retrieve(id, {
-    select: defaultFields,
-    relations: defaultRelations,
+    select: defaultAdminOrdersFields,
+    relations: defaultAdminOrdersRelations,
   })
 
   res.json({ order: data })
 }
+
+export class AdminPostOrdersOrderClaimsClaimReq {
+  @IsArray()
+  @IsOptional()
+  @Type(() => Item)
+  @ValidateNested({ each: true })
+  claim_items: Item[]
+
+  @IsArray()
+  @IsOptional()
+  @ValidateNested({ each: true })
+  @Type(() => ShippingMethod)
+  shipping_methods: ShippingMethod[]
+
+  @IsBoolean()
+  @IsOptional()
+  @Transform(({ value }) => value && value.toString() === "true")
+  @Type(() => Boolean)
+  no_notification?: boolean
+
+  @IsObject()
+  @IsOptional()
+  metadata?: object
+}
+
+class ShippingMethod {
+  @IsString()
+  @IsOptional()
+  id?: string
```

Comment:
Question: Is the id of a shipping method not required if a shipping method is present? (We allow the inclusion of shipping methods to be optional on the Request, but when we provide a shipping method its id should  be included right? Or can we create custom shipping methods this way?)

Reply:
This is optional in the Joi validator, which is why I have also made it optional